conans/client/cmd/export.py

```diff
@@ -233,14 +238,24 @@ def _replace_scm_data_in_conanfile(conanfile_path, scm_data):
                         comments = [line.strip('\n') for line in replace
                                     if line.strip().startswith("#") or not line.strip()]
                         break
-    if len(to_replace) != 1:
+
+    if len(to_replace) > 1:
         raise ConanException("The conanfile.py defines more than one class level 'scm' attribute")
 
     new_text = "scm = " + ",\n          ".join(str(scm_data).split(",")) + "\n"
-    if comments:
-        new_text += '\n'.join(comments) + "\n"
-    content = content.replace(to_replace[0], new_text)
-    content = content if not headers else ''.join(headers) + content
+
+    if len(to_replace) == 0:
+        # SCM exists, but not found in the conanfile, probably inherited from superclass
+        # FIXME: This will inject the lines only the latest class declared in the conanfile
```

Comment:
I think you can check for the `item.bases` of the `item` in line 214:

```python
if isinstance(item, ast.ClassDef):
    if "ConanFile" in [n.id for n in item.bases]:
        ... we are in the only one root ConanFile
```

and we get the prize of the most complex method in Conan 🙈

Reply:
Nope, not possible, in the test example the bases is a function (example extracted directly from the issue)